src/parser.rs

```diff
@@ -2875,3 +2875,54 @@ fn test_parse_type_unit_header_64_ok() {
         otherwise => panic!("Unexpected result: {:?}", otherwise),
     }
 }
+
+/// The `DebugStrings` struct represents the DWARF strings
+/// found in the `.debug_str` section.
+#[derive(Debug, Clone, Copy)]
+pub struct DebugStrings<'input, Endian>
+    where Endian: Endianity
+{
+    debug_str_section: EndianBuf<'input, Endian>,
+}
+
+impl<'input, Endian> DebugStrings<'input, Endian>
+    where Endian: Endianity
+{
+    /// Construct a new `DebugStrings` instance from the data in the `.debug_str`
+    /// section.
+    ///
+    /// It is the caller's responsibility to read the `.debug_str` section and
+    /// present it as a `&[u8]` slice. That means using some ELF loader on
+    /// Linux, a Mach-O loader on OSX, etc.
+    ///
+    /// ```
+    /// use gimli::{DebugStrings, LittleEndian};
+    ///
+    /// # let buf = [0x00, 0x01, 0x02, 0x03];
+    /// # let read_debug_str_section_somehow = || &buf;
+    /// let debug_strings = DebugStrings::<LittleEndian>::new(read_debug_str_section_somehow());
+    /// ```
+    pub fn new(debug_str_section: &'input [u8]) -> DebugStrings<'input, Endian> {
+        DebugStrings { debug_str_section: EndianBuf(debug_str_section, PhantomData) }
+    }
+
+    /// Lookup a string from the `.debug_str` section by DebugStrOffset.
+    ///
+    /// ```
+    /// use gimli::{DebugStrings, DebugStrOffset, LittleEndian};
+    ///
+    /// # let buf = [0x01, 0x02, 0x00];
+    /// # let offset = DebugStrOffset(0);
+    /// # let read_debug_str_section_somehow = || &buf;
+    /// # let debug_str_offset_somehow = || offset;
+    /// let debug_strings = DebugStrings::<LittleEndian>::new(read_debug_str_section_somehow());
+    /// println!("Found string {:?}", debug_strings.get_str(debug_str_offset_somehow()));
+    /// ```
+    pub fn get_str(&self, offset: DebugStrOffset) -> &ffi::CStr {
+        let result = ffi::CStr::from_bytes_until_nul(&self.debug_str_section[offset.0 as usize ..]);
```

Comment:
It's important that the library does not panic for corrupt dwarf data. So you need to check that `offset.0` is in range and return a `ParseResult`.